src/main/java/JordanElimination.java

```diff
@@ -6,15 +10,19 @@
         for (int i = 0; i < numOfEquations; i++) {
             // Replace rows if 0 is found on the diagonal
             if (equation[i][i] == 0) {
-                for (int j = i; j < numOfEquations; j++) {
-                    if (equation[j][i] != 0) {
-                        for (int k = 0; k < numOfEquations + 1; k++) {
-                            double temp = equation[j][k];
-                            equation[j][k] = equation[i][k];
-                            equation[i][k] = temp;
-                        }
+                double maxValue = 0;
+                int maxIndex = 0;
+                for (int j = 0; j < numOfEquations; j++) {
+                    if (maxValue < abs(equation[j][i])) {
+                        maxValue = abs(equation[j][i]);
+                        maxIndex = j;
                     }
                 }
+                for (int k = 0; k < numOfEquations + 1; k++) {
+                    double temp = equation[maxIndex][k];
+                    equation[maxIndex][k] = equation[i][k];
+                    equation[i][k] = temp;
+                }
```

Comment:
You can simplify it to:
```suggestion
                double[] temp = equation[maxIndex];
                equation[maxIndex] = equation[i];
                equation[i] = temp;
```